java/src/main/java/dojo/supermarket/model/ProductQuantity.java

```diff
@@ -0,0 +1,20 @@
+package dojo.supermarket.model;
+
+public class ProductQuantity {
+
+    private final Product product;
+    private final double quantity;
```

Comment:
I would also recommend using BigDecimal to avoid double precision issue (https://www.baeldung.com/java-double-precision-issue)
```suggestion
    private final BigDecimal quantity;
```